src/datachain/lib/listing.py

```diff
@@ -90,6 +91,10 @@ def _isfile(client: "Client", path: str) -> bool:
     Returns True if uri points to a file
     """
     try:
+        if "://" in path:
```

Comment:
why is this needed, btw?

Reply:
This is because when the s3 or other schema was in uppercase, the process would fail since it didn't match the specified regex. Its just that we swallowed the exception in test cases previously.

Reply:
Could you point me to those tests that use uppercase scheme?

Reply:
https://github.com/iterative/datachain/blob/60aed77e41e88c44117568011d21aabdbf0176f7/tests/func/test_ls.py#L78

Reply:
this still very weird place to do this kind of transformation ... how was it handled by ls downstream? Okay, it's swallowed here means we are losing some optimization - by `ls` still worked right? where do we handle this usually?